pkg/webreg/webreg.go

```diff
@@ -328,6 +333,195 @@ const templateDefinitions = `
+Steps are run in the order that they are written. Steps specified by a chain
+can be either references and other chains. Example of a chain:
+</p>
+
+<pre>
+chain:
+  as: ipi-deprovision
+  steps:
+  - chain: gather
+  - ref: ipi-deprovision-deprovision
+  documentation: |-
+    The IPI deprovision step chain contains all the individual steps necessary to deprovision an OpenShift cluster.
+</pre>
+
+<h2>Workflow:</h2>
+<p>
+A workflow is the highest level component of the step registry. It is almost
+identical to the syntax of the ci-operator config for multistage tests and
+defines an entire test from start to finish. It has 4 basic components: a
+cluster_type string (eg: <code>aws</code>, <code>azure4</code>,
+<code>gcp</code>), and 3 chains: <code>pre</code>, <code>test</code>, and
+<code>post</code>. The <code>pre</code> chain is intended to be used to set
+up a testing environment (such as creating a test cluster), the
+<code>test</code> chain is intended to contain all tests that a job wants to
+run, and the <code>post</code> chain is intended to be used to clean up any
+resources created/used by the test. If a step in <code>pre</code> or
+<code>test</code> fails, all pending <code>pre</code> and <code>test</code>
+steps are skipped and all <code>post</code> steps are run to ensure that
+resources are properly cleaned up. This is an example of a workflow config:
+</p>
+
+<pre>
+workflow:
+  as: origin-e2e
+  steps:
+    pre:
+    - ref: ipi-conf
+    - chain: ipi-install
+    test:
+    - ref: origin-e2e-test
+    post:
+    - chain: ipi-deprovision
+  documentation: |-
+	The Origin E2E workflow executes the common end-to-end test suite.
+</pre>
+
+<h2>CI-Operator Test Config:</h2>
+<p>
+The CI-Operator test config syntax for multistage tests is very similar to
+the registry workflow syntax. The main differences are that the ci-operator
+config does not have a <code>documentation</code> field, and the ci-operator
+config can specify a workflow to use. Also, the <code>cluster_profile</code>,
+<code>pre</code>, <code>test</code>, and <code>post</code> fields are under a
+<code>steps</code> field instead of <code>workflow</code>. Here is an example
+of the <code>tests</code> section of a ci-operator config using the
+multistage test design:
+</p>
+
+<pre>
+tests:
+- as: e2e-steps
+  commands: ""
+  steps:
+    cluster_profile: aws
+    workflow: origin-e2e
+</pre>
+
+<p>
+In this example, the ci-operator config simply specifies the desired cluster
+profile and the <code>origin-e2e</code> workflow shown in the example for the
+<code>Workflow</code> section above.
+</p>
+
+<p>
+Since the ci-operator-config and workflows share the same fields, it is
+possible to override fields specified in a workflow. In cases where both the
+workflow and a ci-operator config specify the same field, the ci-operator config’s
+field has priority (i.e. the value from the ci-operator config is used).
```

Comment:
IMO this could use an example too.

Reply:
Added an example for that. Also added comments for fields and syntax highlighting.